### PR TITLE
oclif migration

### PR DESCRIPTION
**Pull request by sachinkalecs**

### CODE REVIEW

**Comment by netrajpatel** [comment]
@sachinkalecs Resolve conflicts and security issues in this PR

File: packages/contentstack-auth/README.md [hunk 1]
```diff
@@ -17,8 +17,8 @@ It is Contentstack’s CLI plugin to perform authentication-related activities.
 $ npm install -g @contentstack/cli-auth
 $ csdx COMMAND
 running command...
-$ csdx (-v|--version|version)
-@contentstack/cli-auth/1.0.0 darwin-x64 node-v16.15.1
+$ csdx (--version)
+@contentstack/cli-auth/1.0.0 linux-x64 node-v16.14.2
```
Comment (netrajpatel) [inline packages/contentstack-auth/README.md:21]: Why are we not using the latest LTS here?
Why are we bringing down the node version from 16.15.1 to 16.14.2, I can see latest LTS is at 16.17.0?

File: packages/contentstack-bootstrap/README.md [hunk 2]
```diff
@@ -14,8 +14,8 @@ This means that all the required steps such as stack, environment, and content t
 $ npm install -g @contentstack/cli-cm-bootstrap
 $ csdx COMMAND
 running command...
-$ csdx (-v|--version|version)
-@contentstack/cli-cm-bootstrap/1.1.0 darwin-x64 node-v16.15.1
+$ csdx (--version)
+@contentstack/cli-cm-bootstrap/1.1.0 linux-x64 node-v16.14.2
```
Comment (netrajpatel) [inline packages/contentstack-bootstrap/README.md:18]: Node version change to latest LTS

File: packages/contentstack-config/README.md [hunk 3]
```diff
@@ -15,8 +15,8 @@ It is Contentstack’s CLI plugin to perform authentication-related activities.
 $ npm install -g @contentstack/cli-config
 $ csdx COMMAND
 running command...
-$ csdx (-v|--version|version)
-@contentstack/cli-config/1.0.0 darwin-x64 node-v16.15.1
+$ csdx (--version)
+@contentstack/cli-config/1.0.0 linux-x64 node-v16.14.2
```
Comment (netrajpatel) [inline packages/contentstack-config/README.md:19]: Node version change to latest LTS

File: packages/contentstack-auth/package.json [hunk 4]
```diff
@@ -21,16 +21,13 @@
...
-    "@oclif/config": "^1.18.3",
     "chalk": "^4.0.0",
     "debug": "^4.1.1",
     "inquirer": "^7.1.0",
     "winston": "^3.7.2"
   },
   "devDependencies": {
-    "@oclif/dev-cli": "^1.22.2",
-    "@oclif/plugin-help": "^5.1.12",
+    "oclif": "^3",
```
Comment (netrajpatel) [inline packages/contentstack-auth/package.json:30]: Can we use the latest oclif verion [3.1.2] ?

File: packages/contentstack-bootstrap/package.json [hunk 5]
```diff
@@ -10,26 +10,23 @@
...
-    "@oclif/command": "^1.8.16",
-    "@oclif/config": "^1.18.3",
     "inquirer": "^8.0.0",
     "mkdirp": "^1.0.4",
     "tar": "^6.0.5"
   },
   "devDependencies": {
-    "@oclif/dev-cli": "^1.22.2",
-    "@oclif/plugin-help": "^5.1.12",
+    "oclif": "^3",
```
Comment (netrajpatel) [inline packages/contentstack-bootstrap/package.json:29]: Can we use the latest oclif verion [3.1.2] ?

File: packages/contentstack-bulk-publish/package.json [hunk 6]
```diff
@@ -8,17 +8,15 @@
...
     "lodash": "^4.17.15",
     "nock": "^13.0.2",
-    "winston": "^3.7.2",
-    "inquirer": "^8.0.0"
+    "winston": "^3.7.2"
   },
   "devDependencies": {
-    "@oclif/dev-cli": "^1.22.2",
-    "@oclif/plugin-help": "^5.1.12",
+    "oclif": "^3",
```
Comment (netrajpatel) [inline packages/contentstack-bulk-publish/package.json:19]: Can we use the latest oclif verion [3.1.2] ?

File: packages/contentstack-seed/package.json [hunk 7]
```diff
@@ -18,8 +16,7 @@
     "tslib": "^1.13.0"
   },
   "devDependencies": {
-    "@oclif/dev-cli": "^1.22.2",
-    "@oclif/plugin-help": "^5.1.12",
+    "oclif": "^3",
```
Comment (netrajpatel) [inline packages/contentstack-seed/package.json:19]: Can we use the latest oclif verion [3.1.2] ?

File: packages/contentstack-utilities/package.json [hunk 8]
```diff
@@ -27,7 +27,7 @@
   "license": "MIT",
   "dependencies": {
     "@contentstack/management": "^1.3.0",
-    "@oclif/core": "^1.8.2",
+    "@oclif/core": "^1.13.10",
```
Comment (netrajpatel) [inline packages/contentstack-utilities/package.json:30]: we can install the oclif and include the core package from oclif, why we need to install separate package if we already have oclif package installed at root.

File: packages/contentstack/package.json [hunk 9]
```diff
@@ -49,8 +46,7 @@
     "winston": "^3.7.2"
   },
   "devDependencies": {
-    "@oclif/dev-cli": "^1.22.2",
-    "@oclif/plugin-help": "^5.1.12",
+    "oclif": "^3",
     "@oclif/test": "^1.2.8",
```
Comment (netrajpatel) [inline packages/contentstack/package.json:50]: Do we still need to install oclif/test separately if we already have oclif installed above?

**Review by netrajpatel — CHANGES_REQUESTED** [review]

File: packages/contentstack-auth/README.md (hunk 1 above)
Comment (sachinkalecs) [reply packages/contentstack-auth/README.md]: done

File: packages/contentstack-bootstrap/README.md (hunk 2 above)
Comment (sachinkalecs) [reply packages/contentstack-bootstrap/README.md]: done

File: packages/contentstack-config/README.md (hunk 3 above)
Comment (sachinkalecs) [reply packages/contentstack-config/README.md]: done

File: packages/contentstack-auth/package.json (hunk 4 above)
Comment (sachinkalecs) [reply packages/contentstack-auth/package.json]: done

File: packages/contentstack-bootstrap/package.json (hunk 5 above)
Comment (sachinkalecs) [reply packages/contentstack-bootstrap/package.json]: done

File: packages/contentstack-seed/package.json (hunk 7 above)
Comment (sachinkalecs) [reply packages/contentstack-seed/package.json]: done

File: packages/contentstack-bulk-publish/package.json (hunk 6 above)
Comment (sachinkalecs) [reply packages/contentstack-bulk-publish/package.json]: done

File: packages/contentstack-utilities/package.json (hunk 8 above)
Comment (sachinkalecs) [reply packages/contentstack-utilities/package.json:30]: As oclif dev dependency package is different and does not include @oclif/core package we have install @oclif/core package.

File: packages/contentstack/package.json (hunk 9 above)
Comment (sachinkalecs) [reply packages/contentstack/package.json:50]: As oclif dev dependency package is different and does not include @oclif/test package we have install @oclif/test package.

**Comment by sachinkalecs** [comment]
Will reopen once tested by QA

**Pull request closed without merging**